src/main/java/uk/gov/hmcts/reform/professionalapi/configuration/resolver/UserIdArgumentResolver.java

```diff
@@ -35,6 +42,11 @@ public Object resolveArgument(
 
         HttpServletRequest request
                 = (HttpServletRequest) nativeWebRequest.getNativeRequest();
+
+        if (!deleteOrganisationEnabled && request.getMethod().equals("DELETE")) {
+
+            throw new AccessDeniedException(ERROR_MESSAGE_403_FORBIDDEN);
+        }

```

Comment:
I dont think this is the right place to do the feature toggle. to keep things simple and easy to maintain, i would suggest you to add this at the controller level